vrecord

```diff
@@ -925,6 +935,15 @@ if [ -z "${duration}" ] ; then
     read duration
 fi
 
+# confirm legal value for diration
+if [ -n "${duration}" ]; then
+real_regexp="[+-]?([0-9]+|[0-9]+\.[0-9]*|\.[0-9]+)"
+if [[ $duration =~ ^$real_regexp$ ]]; then
+is_number=yes
+else echo "Illegal value for recording time. Input must only be numbers." && exit
+fi
+fi
+
```

Comment:
Thank you for the answer! I did understand that. My remark was meant from the maintainer's perspective: only once the code, called in two different ways.